File: src/integrationTest/java/info/idgst/integration/DigestResourceIntegrationTest.java
package info.idgst.integration;

import info.idgst.digest.Digest;
import info.idgst.digest.DigestRepository;
import info.idgst.digest.reader.JsonDigestReader;
import org.apache.commons.lang3.RandomStringUtils;
import org.apache.http.HttpStatus;
import org.apache.http.client.methods.CloseableHttpResponse;
import org.apache.http.client.methods.HttpGet;
import org.apache.http.entity.ContentType;
import org.apache.http.impl.client.HttpClientBuilder;
import org.junit.Test;
import org.springframework.beans.factory.annotation.Autowired;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.MatcherAssert.assertThat;

/**
 * Integration test for {@link info.idgst.digest.DigestResource}.
 *
 * @author Aliaksei Bahdanau.
 */
public class DigestResourceIntegrationTest extends AbstractIntegrationTest {

    private static final String DIGEST_API_URL = "/api/v1/digests/";

    @Autowired
    private DigestRepository digestRepository;
    @Autowired
    private JsonDigestReader digestReader;


    @Override
    public void before() throws Exception {
        super.before();

        String path = "src/test/resources/digest.json";
        Digest digest = digestReader.readDigest(Files.readAllBytes(Paths.get(path)));

        digestRepository.save(digest);
    }

    @Override
    public void after() throws Exception {
        super.after();

        digestRepository.deleteAll();
    }

    @Test
    public void digestResource_notFound() throws IOException {
        // Setup
        final String name = RandomStringUtils.randomAlphabetic(8);
        final HttpGet request = new HttpGet(protocol + "://" + serverHost + ":" + serverPort + DIGEST_API_URL + name);

        // Run
        final CloseableHttpResponse httpResponse = HttpClientBuilder.create().build().execute(request);

        // Verify
        assertThat(httpResponse.getStatusLine().getStatusCode(), is(HttpStatus.SC_NOT_FOUND));
    }

    @Test
    public void digestReturnedAsJson() throws IOException {
        String jsonMimeType = "application/json";
        final HttpGet request = new HttpGet(protocol + "://" + serverHost + ":" + serverPort + DIGEST_API_URL);

        // Run
        final CloseableHttpResponse httpResponse = HttpClientBuilder.create().build().execute(request);

        // Verify
        final String mimeType = ContentType.getOrDefault(httpResponse.getEntity()).getMimeType();
        assertThat(mimeType, is(jsonMimeType));
    }
}

File: src/integrationTest/kotlin/info/devbug/integration/digest/DigestResourceIntegrationTest.kt
package info.devbug.integration.digest

import info.devbug.digest.repository.DigestRepository
import info.devbug.digest.util.DigestReader
import info.devbug.digest.util.JsonDigestReader
import info.devbug.integration.AbstractIntegrationTest
import org.apache.commons.lang3.RandomStringUtils
import org.apache.http.HttpStatus
import org.apache.http.client.methods.HttpGet
import org.apache.http.entity.ContentType
import org.apache.http.impl.client.HttpClientBuilder
import org.junit.After
import org.junit.Before
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired

/**
 * Integration test for DigestResource
 *
 * @author Aliaksei Bahdanau.
 */
class DigestResourceIntegrationTest : AbstractIntegrationTest() {

    val digestApiUrl = "/v1/digests/"

    lateinit private var digestReader: DigestReader

    @Autowired lateinit private var digestRepository: DigestRepository

    @Before
    fun setUp() {
        digestReader = JsonDigestReader()

        val digest = digestReader.readDigest("src/test/resources/digest.json")

        digestRepository.save(digest)
    }

    @After
    fun cleanUp() {
        digestRepository.deleteAll()
    }

    @Test
    @Throws(Exception::class)
    fun `resource doesn't exist`() {
        // Given
        val name = RandomStringUtils.randomAlphabetic(8)
        val request = HttpGet("$protocol://$serverHost:$serverPort$digestApiUrl$name")

        // When
        val httpResponse = HttpClientBuilder.create().build().execute(request)

        // Then
        assert(httpResponse.statusLine.statusCode == HttpStatus.SC_NOT_FOUND)
    }

    @Test
    @Throws(Exception::class)
    fun `all digest returned as JSON`() {
        // Given
        val request = HttpGet("$protocol://$serverHost:$serverPort$digestApiUrl")
        val jsonMimeType = "application/json"

        // When
        val httpResponse = HttpClientBuilder.create().build().execute(request)

        // Then
        assert(httpResponse.statusLine.statusCode == HttpStatus.SC_OK)

        val mimeType: String = ContentType.getOrDefault(httpResponse.entity).mimeType
        assert(jsonMimeType == mimeType)
    }
}